signer/src/testing/transaction_signer.rs

```diff
@@ -229,6 +276,50 @@ where
         context_window_block_hashes
     }
 
+    async fn extract_stacks_context_window_block_hashes(
+        &self,
+        storage: &S,
+    ) -> Vec<model::StacksBlockHash> {
+        let canoncial_tip_block_hash = storage
+            .get_bitcoin_canonical_chain_tip()
+            .await
+            .unwrap()
+            .expect("found no canonical chain tip");
+
+        let chain_tip = storage
+            .get_bitcoin_block(&canoncial_tip_block_hash)
+            .await
+            .expect("failed to get bitcoin block")
+            .expect("missing block");
+
+        let stacks_chain_tip = futures::stream::iter(chain_tip.confirms)
```

Comment:
Great question, and yes we can. Essentially, any stacks block referenced in a leader block commit transaction within the bitcoin block is confirmed by that bitcoin block. We don't need to run any complicated sortition logic. Only the much simpler transaction parsing logic for this.

However, one important note is that we cannot assume that we know of a stacks block just because it is confirmed. In this test utility that happens to be guaranteed right now, but in the production logic we should always assume that trying to look up a block based on it's block ID could yield an empty result.

Reply:
LeaderBlockCommit reference: https://github.com/stacksgov/sips/blob/main/sips/sip-001/sip-001-burn-election.md#leader-block-commit - it has been updated a bit in subsequent SIPs but as I understand it the basic principle is the same, even in Nakamoto.